tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -4651,6 +4652,159 @@ public function downloadPreviewOfFiles(string $user, string $path, string $width
 		$this->setResponse($response);
 	}
 
+	/**
+	 * @When user :user downloads the preview of shared resource :path with width :width and height :height using the WebDAV API
+	 *
+	 * @param string $user
+	 * @param string $path
+	 * @param string $width
+	 * @param string $height
+	 *
+	 * @return void
+	 */
+	public function userDownloadsThePreviewOfSharedResourceWithWidthAndHeightUsingTheWebdavApi(string $user, string $path, string $width, string $height): void {
+		if ($this->getDavPathVersion() === 3) {
+			$this->setResponse($this->userDownloadsOrUploadsSharedResource($user, $path, 'GET', $width, $height));
+		} else {
+			$this->setResponse($this->downloadPreviews($user, $path, null, $width, $height));
+		}
```

Comment:
If we move `userDownloadsOrUploadsSharedResource` inside of `downloadPreviews`, for `spaces` dav version, `userDownloadsOrUploadsSharedResource` will be executed every time because of the if condition and we cannot differentiate between whether the resource to be previewed is shared resource or not.